docs/design/CONFIG_API.md

```diff
@@ -0,0 +1,249 @@
+Configuration API Design
```

Comment:
I think it'd be helpful to use the same terms across different docs for clarity:

```suggestion
Static Configuration API Design
```

Reply:
@skriss This API intends to cover static and dynamic config, see the latest commit for additional details.

Reply:
@skriss the latest commit uses a variation of this feedback by having "Control Plane API" and "Data Plane API" sections. I prefer this approach since we may express control plane configuration dynamically in the future and vice versa for data plane config.